drivers/sdp3x/sdp3x.c

```diff
@@ -0,0 +1,473 @@
+    return _SDP3x_convert_to_celsius(data[1]);
+}
+
+/**
+ *      intern function to read differential pressure
+ *      @param  dev      device
+ *      @return Pressure value in pascals
+ */
+double _SDP3x_read_pressure(const sdp3x_t *dev)
+{
+    int16_t data[3];
+    uint8_t ret = _SDP3x_read_data(dev, data);
+
+    if (ret != 0) {
+        DEBUG("[SDP3x] Error reading date\n");
+        return ret;
+    }
+    return _SDP3x_convert_to_pascal(dev, data[0]);
+}
+
+/** Read measurements
+ *
+ *      Data consists of:
+ *      2 byte Differential Pressure,
+ *      1 byte CRC
+ *      2 byte Temperature,
+ *      1 byte CRC
+ *      2 byte Scale Factor differential pressure
+ *      1 byte CRC
+ *
+ *      @param addr   Address byte 1 = 0x21, 2 = 0x22, 3 = 0x23,
+ *                    else = Last address
+ *      @param data   Data will be stored here
+ *
+ *      @return       0 if data could be read, 1 if CRC-Error
+ */
+static int8_t _SDP3x_read_data(const sdp3x_t *dev, int16_t *data)
+{
+    int ret = 0;
+    uint8_t *readData = calloc(9, sizeof(uint8_t));
+
+    i2c_acquire(DEV_I2C);
+    ret = i2c_read_bytes(DEV_I2C, DEV_ADDR, readData, 9, 0);
+    i2c_release(DEV_I2C);
+
+    if (ret != 0) {
+        DEBUG("[SDP3x] read_data: ret %i\n", ret);
+        free(readData);
+        return ret;
+    }
+    data[2] = (readData[6] << 8) + readData[7];
+    data[1] = (readData[3] << 8) + readData[4];
+    data[0] = (readData[0] << 8) + readData[1];
+
+    uint8_t correct = 1;
+
+    correct &= _checkCRC(data[2], readData[8]);
+    correct &= _checkCRC(data[1], readData[5]);
+    correct &= _checkCRC(data[0], readData[2]);
+
+    if (correct == 0) {
+        DEBUG("[SDP3x] read_data: CRC-Failure\n");
+        free(readData);
+        return SDP3x_CRCERROR;
+    }
+
+    free(readData);
+    return 0;
+}
+
+/**
+ *      intern method to convert sensor value to pascal
+ *      @param  dev      device
+ *      @param  value    raw sensor value
+ *      @return pascal value
+ */
+static double _SDP3x_convert_to_pascal(const sdp3x_t *dev, int16_t value)
+{
+    double div = 1.0;
+
+    if (DEV_MODEL == 1) {
+        div = 60.0;
+    }
+    else {
+        div = 240.0;
+    }
+    return (((double)value) / div);
+}
+
+/**
+ *      intern method to convert sensor value to inches h2o
+ *      @param  dev         device
+ *      @param  value       raw sensor value
+ *      @return inches h2o value
+ */
+static double _SDP3x_convert_to_inches_h2o(const sdp3x_t *dev, int16_t value)
+{
+    double div = 1.0;
+
+    if (DEV_MODEL == 1) {
+        div = 14945.0;
+    }
+    else {
+        div = 59780.0;
+    }
+    return (((double)value) / div);
+}
+
+/**
+ *      intern method to convert sensor value to celsius
+ *      @param  dev     device
+ *      @param  value   raw sensor value
+ *      @return celsius value
+ */
+static double _SDP3x_convert_to_celsius(int16_t value)
+{
+    double div = 200;
+
+    return ((double)value / div);
+}
```

Comment:
So the raw sensor value will be 200 * °C, right?

Then `SDP3x_read_single_temperature` could just return m°C without any loss of precision.